### PR TITLE
Hookup 2FA backup code UI to live data

### PR DESCRIPTION
**Pull request by dblandin**
- Query for 2FA backup codes when requested
- Create new backup codes when requested and and display

### Screen Recording

Peek 2020-04-23 17-10

### CODE REVIEW

File: src/Components/UserSettings/TwoFactorAuthentication/index.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          }
+        }
+
+        smsSecondFactors: secondFactors(kinds: [sms]) {
+          ... on SmsSecondFactor {
+            formattedPhoneNumber
+          }
+        }
+
+        ...BackupSecondFactor_me @relay(mask: false)
```
Comment (dblandin) [inline src/Components/UserSettings/TwoFactorAuthentication/index.tsx:65]: I'm not sure if `mask: false` is the best way to handle this. Without it, the alias that I'm defining in the fragment isn't available in the TwoFactorAuthentication_me type...


```
src/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactor/index.tsx:53:21 - error TS2339: Property 'backupSecondFactors' does not exist on type 'TwoFactorAuthentication_me'.

53                 {me.backupSecondFactors.length} remaining
```

File: src/Components/UserSettings/TwoFactorAuthentication/index.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      }
+    `,
+  }
+)
+
+export const TwoFactorAuthenticationQueryRenderer = () => {
```
Comment (dblandin) [inline src/Components/UserSettings/TwoFactorAuthentication/index.tsx:71]: This rename will require another update in Force where we're stitching the component in.

File: src/Components/UserSettings/TwoFactorAuthentication/__tests__/TwoFactorAuthentication.test.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      env.mutations.useResultsOnce(BackupSecondFactorsMutationResponse)
+      env.mockQuery.mockImplementation(
+        createMockFetchQuery({
+          mockData: AppEnabledWithBackupCodesQueryResponse,
+        })
+      )
+
+      await page.clickBackupSetupButton()
+
+      setTimeout(() => {
```
Comment (dblandin) [inline src/Components/UserSettings/TwoFactorAuthentication/__tests__/TwoFactorAuthentication.test.tsx:78]: This seems wrong but I can't get these specs to pass in CI without it. Passes fine locally. /shrug

**Comment by dblandin** [comment]
Going to go ahead and merge but happy to incorporate any feedback in subsequent PRs.

**Comment by artsyit** [comment]
:rocket: PR was released in [`v26.27.0`](https://github.com/artsy/reaction/releases/tag/v26.27.0) :rocket:

**Pull request merged**